### PR TITLE
Fix no reads for certain barcode issue

### LINKED ISSUE #63: if there are NO reads with a certain barcode, give error message (and mail)
The check whether barcode is 0.0% is done in the Demultiplexing.sh but the error is produced in the Illumina2GafBarcode step (before the Demultiplexing.sh step)

### PR DESCRIPTION
**Pull request by marieke-bijlsma**
Fixes #63

### CODE REVIEW

File: protocols/Illumina2GafFastQ.sh [hunk 1]
```diff
@@ -35,30 +36,35 @@ do
...
+				cd "${runResultsDir}"
+				VAR1=${sequencingStartDate}_${sequencer}_${run}_${flowcell}_L${lane}_${barcode_combined[sampleNumber]}.fq.gz
+				perl -pi -e "s|lane${lane}_${barcode_combined[sampleNumber]}_S[0-9]+_L00${lane}_R1_001.fastq.gz|$VAR1|" $VAR1.md5
 
-			md5sum -c ${sequencingStartDate}_${sequencer}_${run}_${flowcell}_L${lane}_${barcode_combined[sampleNumber]}.fq.gz.md5
-	fi
+				md5sum -c ${sequencingStartDate}_${sequencer}_${run}_${flowcell}_L${lane}_${barcode_combined[sampleNumber]}.fq.gz.md5
+			else
+				echo -e "No reads detected with: lane${lane}_${barcode_combined[sampleNumber]}_S[0-9]*_L00${lane}_R1_001.fastq.gz.\n\nCheers GCC" | mail -s "No reads with certain barcode" "${mailingList}"
+			fi
```
Comment (RoanKanninga) [inline protocols/Illumina2GafFastQ.sh]: I think we need an exit 1 here also? The pipeline will crash later on when there are no reads for that particular barcode present..

File: protocols/Illumina2GafFastQ.sh [hunk 2]
```diff
@@ -101,37 +112,39 @@ then
 		cd "${intermediateDir}"
 		md5sum Undetermined_S[0-9]*_L00${lane}_R1_001.fastq.gz >  ${runResultsDir}/${sequencingStartDate}_${sequencer}_${run}_${flowcell}_L${lane}_DISCARDED.fq.gz.md5
 
-                cp Undetermined_S[0-9]*_L00${lane}_R1_001.fastq.gz ${runResultsDir}/${sequencingStartDate}_${sequencer}_${run}_${flowcell}_L${lane}_DISCARDED.fq.gz
+		cp Undetermined_S[0-9]*_L00${lane}_R1_001.fastq.gz ${runResultsDir}/${sequencingStartDate}_${sequencer}_${run}_${flowcell}_L${lane}_DISCARDED.fq.gz
+
+		cd  "${runResultsDir}"
 
-                cd "${runResultsDir}"
+		VAR1=${sequencingStartDate}_${sequencer}_${run}_${flowcell}_L${lane}_DISCARDED.fq.gz
```
Comment (RoanKanninga) [inline protocols/Illumina2GafFastQ.sh]: missing quotes ;)

File: protocols/Illumina2GafFastQ.sh [hunk 3]
```diff
@@ -101,37 +112,39 @@ then
...
+		VAR2=${sequencingStartDate}_${sequencer}_${run}_${flowcell}_L${lane}_DISCARDED_2.fq.gz
 
-                perl -pi -e "s|Undetermined_S[0-9]+_L00${lane}_R1_001.fastq.gz|$VAR1|" $VAR1.md5
-                perl -pi -e "s|Undetermined_S[0-9]+_L00${lane}_R2_001.fastq.gz|$VAR2|" $VAR2.md5
+		perl -pi -e "s|Undetermined_S[0-9]+_L00${lane}_R1_001.fastq.gz|$VAR1|" $VAR1.md5
+		perl -pi -e "s|Undetermined_S[0-9]+_L00${lane}_R2_001.fastq.gz|$VAR2|" $VAR2.md5
 
-                md5sum -c ${sequencingStartDate}_${sequencer}_${run}_${flowcell}_L${lane}_DISCARDED_1.fq.gz.md5
-                md5sum -c ${sequencingStartDate}_${sequencer}_${run}_${flowcell}_L${lane}_DISCARDED_2.fq.gz.md5
+		md5sum -c ${sequencingStartDate}_${sequencer}_${run}_${flowcell}_L${lane}_DISCARDED_1.fq.gz.md5
```
Comment (RoanKanninga) [inline protocols/Illumina2GafFastQ.sh]: quotes

File: protocols/Illumina2GafFastQ.sh [hunk 4]
```diff
@@ -101,37 +112,39 @@ then
...
+		md5sum -c ${sequencingStartDate}_${sequencer}_${run}_${flowcell}_L${lane}_DISCARDED_2.fq.gz.md5
 	fi
+
 else
-		echo "There can't be discarded reads because the Barcode is set to None"
+	echo "There can't be discarded reads because the Barcode is set to None"
 fi
 
 cd $OLDDIR
```
Comment (RoanKanninga) [inline protocols/Illumina2GafFastQ.sh]: curly brackets + quotes

File: protocols/Illumina2GafFastQ.sh (hunk 2 above)
Comment (marieke-bijlsma) [reply protocols/Illumina2GafFastQ.sh]: Nice, you're commenting on your own code :p

**Pull request merged**